tools/generator/src/Extensions/XCScheme+Extensions.swift

```diff
@@ -0,0 +1,210 @@
+                customLLDBInitFile: XCSchemeConstants.customLLDBInitFile
+            )
+        }
+
+        let profileAction: XCScheme.ProfileAction?
+        if let profileActionInfo = schemeInfo.profileActionInfo {
+            profileAction = .init(profileActionInfo: profileActionInfo)
+        } else {
+            profileAction = .init(
+                buildableProductRunnable: nil,
+                buildConfiguration: XCSchemeConstants.defaultBuildConfigurationName
+            )
+        }
+
+        self.init(
+            name: schemeInfo.name,
+            lastUpgradeVersion: XCSchemeConstants.defaultLastUpgradeVersion,
+            version: XCSchemeConstants.lldbInitVersion,
+            buildAction: buildAction,
+            testAction: testAction,
+            launchAction: launchAction,
+            profileAction: profileAction,
+            analyzeAction: .init(analyzeActionInfo: schemeInfo.analyzeActionInfo),
+            archiveAction: .init(archiveActionInfo: schemeInfo.archiveActionInfo),
+            wasCreatedForAppExtension: schemeInfo.wasCreatedForAppExtension ? true : nil
+        )
+    }
+}
+
+extension XCScheme.BuildAction {
+    convenience init(
+        buildMode: BuildMode,
+        buildActionInfo: XCSchemeInfo.BuildActionInfo
+    ) throws {
+        self.init(
+            buildActionEntries: buildActionInfo.targetInfos.buildActionEntries,
+            preActions: try buildActionInfo.targetInfos.buildPreActions(buildMode: buildMode),
+            parallelizeBuild: true,
+            buildImplicitDependencies: true
+        )
+    }
+}
+
+extension XCScheme.BuildAction.Entry {
+    convenience init(withDefaults buildableReference: XCScheme.BuildableReference) {
+        self.init(
+            buildableReference: buildableReference,
+            buildFor: [
+                .running,
+                .testing,
+                .profiling,
+                .archiving,
+                .analyzing,
+            ]
+        )
+    }
+}
+
+extension XCScheme.ExecutionAction {
+    /// Initialize the output file for Build with Bazel mode.
+    static let initBazelBuildOutputGroupsFile = XCScheme.ExecutionAction(
+        scriptText: #"""
+mkdir -p "${BAZEL_BUILD_OUTPUT_GROUPS_FILE%/*}"
+if [[ -s "$BAZEL_BUILD_OUTPUT_GROUPS_FILE" ]]; then
+    rm "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+fi
+
+"""#,
+        title: "Initialize Bazel Build Output Groups File"
+    )
+
+    /// Create an `ExecutionAction` that builds with Bazel.
+    convenience init(
+        buildFor buildableReference: XCScheme.BuildableReference,
+        name: String,
+        hostIndex: Int?,
+        buildMode: BuildMode
+    ) {
+        let prefix = buildMode.buildOutputGroupPrefix
+        let hostTargetOutputGroup: String
+        if let hostIndex = hostIndex {
+            hostTargetOutputGroup = #"""
+echo "\#(prefix) $BAZEL_HOST_TARGET_ID_\#(hostIndex)" \#
+>> "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+"""#
```

Comment:
This is missing a newline if we have a host:
```suggestion
>> "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"

"""#
```